vars/assembleUpload.groovy

```diff
@@ -1,13 +1,8 @@
+
 void call(Map args = [:]) {
 
     lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
 
     assembleManifest(args)
     uploadArtifacts(args)
-
-    def inputManifest = lib.jenkins.InputManifest.new(readYaml(file: args.manifest))
-    def buildManifest = lib.jenkins.BuildManifest.new(readYaml(file: args.manifest))
-    String artifactUrl = inputManifest.getPublicDistUrl("${PUBLIC_ARTIFACT_URL}", "${JOB_NAME}", "${BUILD_NUMBER}", buildManifest.build.platform, buildManifest.build.architecture)
-    echo "Setting env.\"ARTIFACT_URL_${buildManifest.build.platform}_${buildManifest.build.architecture}\"=${artifactUrl}"
```

Comment:
This shouldn't be deleted, it creates the part of the build notification with instructions on where the build is located.

Reply:
From my understanding this isn't used by the build notifications - instead, messages are added in [uploadArtifacts](https://github.com/opensearch-project/opensearch-build/blob/main/vars/uploadArtifacts.groovy#L32) which is called in the build steps via [buildAssembleUpload](https://github.com/opensearch-project/opensearch-build/blob/main/jenkins/opensearch/distribution-build.jenkinsfile#L101). Then, the message with the links to the manifests are added in the notification's `extra` field [here](https://github.com/opensearch-project/opensearch-build/blob/main/jenkins/opensearch/distribution-build.jenkinsfile#L151).

But, these env vars are still used in [buildDockerImage](https://github.com/opensearch-project/opensearch-build/blob/main/vars/buildDockerImage.groovy#L9). These can be removed though, and we could set them explicitly in the pipeline itself (we already derive the value [here](https://github.com/ohltyler/opensearch-build/blob/add-to-distribution-build/jenkins/opensearch/distribution-build.jenkinsfile#L117)). Then they can be passed explicitly to `buildDockerImage()` instead of relying on an env var being set. What do you think?

Reply:
Ah, thanks for checking in on that.

I would prefer moving to parameters to remove these 'hidden' ENV variables

Reply:
Totally agreed, I'll go ahead with that approach